runtime-sdk/modules/evm/src/raw_tx.rs

```diff
@@ -99,9 +100,13 @@ pub fn decode(
             let sig_recid = k256::ecdsa::RecoveryId::new(eth_tx.odd_y_parity, false);
             let message = ethereum::EIP1559TransactionMessage::from(eth_tx);
 
-            // Base fee is zero. Allocate only priority fee.
-            let resolved_gas_price =
-                std::cmp::min(message.max_fee_per_gas, message.max_priority_fee_per_gas);
+            let min_gas_price = min_gas_price.into();
```

Comment:
I would rename a bit, to be in line with the EIP.

```rust
            let base_fee_per_gas = min_gas_price.into();
            let priority_fee_per_gas = std::cmp::min(
                message.max_priority_fee_per_gas,
                message.max_fee_per_gas.saturating_sub(base_fee_per_gas),
            );
            let effective_gas_price = priority_fee_per_gas.saturating_add(base_fee_per_gas);
```

Do we need to error if the following doesn't hold?
```rust
balance >= gas_limit * max_fee_per_gas
max_fee_per_gas >= max_priority_fee_per_gas
max_fee_per_gas >= block.base_fee_per_gas
```

Reply:
We don't need to check these things here as it will be handled by the core module during transaction checks. This is transaction decoding so it should do as little as possible.

Reply:
I think that this decoding hides `max_priority_fee_per_gas` and `max_fee_per_gas` from the core, so the second inequality cannot be checked. So in this case, EIP would reject the transaction and we don't. No sure, if this is relevant, though.

Reply:
Fixed.